.github/actions/bump-version/action.yml

```diff
@@ -21,11 +24,24 @@ runs:
         git fetch origin
         git checkout ${{ inputs.target_branch }}
 
-        # determine current version
-        oldVersion=$(grep "version" gradle.properties  | awk -F= '{print $2}')
-
+        # use current version from input
+        oldVersion=${{ inputs.from_version }}
+        if [ -z $oldVersion ]; then
+          #... or read it from gradle.properties
+          echo "No from_version input was given, will read base version from gradle.properties"
+          oldVersion=$(grep "version" gradle.properties  | awk -F= '{print $2}')
+        fi      
+        
         # read the major, minor, and patch components, consume -SNAPSHOT
         IFS=.- read -r RELEASE_VERSION_MAJOR RELEASE_VERSION_MINOR RELEASE_VERSION_PATCH SNAPSHOT<<<"$oldVersion"
+        INC=0
+        # Compute new snapshot version, do not increment snapshot on non-final releases, e.g. -rc1
+        if [ -z $SNAPSHOT ]; then
+          echo "$oldVersion is a final release version, increase patch for next snapshot"
+          INC=1
+        else
+          echo "$oldVersion is not a final release version (contains \"$SNAPSHOT\"), will not increase patch"
+        fi
 
         # construct the new version
         newVersion="$RELEASE_VERSION_MAJOR.$RELEASE_VERSION_MINOR.$((RELEASE_VERSION_PATCH+1))"-SNAPSHOT
```

Comment:
shouldn't this be `$((RELEASE_VERSION_PATCH+INC))`? as it is, this would always increase the path version, right?

Reply:
you are absolutely right. copy-paste error...